.github/workflows/test.yml

```diff
@@ -41,3 +41,75 @@ jobs:
 
       - name: Run Cargo Tests
         run: cargo test --workspace ${{ matrix.feature-args }}
+
+  cross-platform-test:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-14, macos-14-large]
+        feature-args: ['', '-Funstable-mobile-app']
+        include:
+          - feature-args: ''
+            feature-suffix: ''
+          - feature-args: '-Funstable-mobile-app'
+            feature-suffix: ' (-Funstable-mobile-app)'
+
+    name: Build on macOS 14, Test on macOS 13${{ matrix.feature-suffix }}
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # 4.2.2
+
+      - name: Cache Dependencies
+        uses: swatinem/rust-cache@98c8021b550208e191a6a3145459bfc9fb29c4c0 # 2.8.0
+
+      - name: Build on macOS 14
+        run: cargo build --release --workspace ${{ matrix.feature-args }}
+
+      - name: Upload Build Artifacts
+        uses: actions/upload-artifact@ea165f8d65b6e75b540449e92b4886f43607fa02 # 4.6.2
+        with:
+          name: sentry-cli-macos14-build${{ matrix.feature-suffix }}-${{ matrix.os }}
+          path: target/release/sentry-cli
+          retention-days: 1
+
+  cross-platform-test-runner:
+    needs: cross-platform-test
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-13, macos-13-xlarge, macos-15-large]
+        feature-args: ['', '-Funstable-mobile-app']
+        include:
+          - os: macos-13
+            base-os: macos-14-large
+          - os: macos-13-xlarge
+            base-os: macos-14
+          - os: macos-15-large
+            base-os: macos-14-large
+          - feature-args: ''
+            feature-suffix: ''
+          - feature-args: '-Funstable-mobile-app'
+            feature-suffix: ' (-Funstable-mobile-app)'
+
+    name: Run macOS ${{ matrix.os }} ${{ matrix.feature-suffix }}
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # 4.2.2
+
+      - name: Download Build Artifacts
+        uses: actions/download-artifact@d3f86a106a0bac45b974a628896c90dbdf5c8093 # 4.3.0
+        with:
+          name: sentry-cli-macos14-build${{ matrix.feature-suffix }}-${{ matrix.base-os }}
+          path: ./bin
+
+      - name: Make Binary Executable
+        run: chmod +x ./bin/sentry-cli
+
+      - name: Test Binary Compatibility
+        run: |
+          # Ensure the cli runs
+          ./bin/sentry-cli --help
```

Comment:
If we will make this feature ARM-specific, we should have test coverage for both the ARM-binary and the x86_64-binary; however, I would prefer that we try to keep it simpler than this (i.e. depending on fewer runner types, and perhaps only testing on the system we build for).